### PR TITLE
Bug 2041475: Add function names to components which are wrapped in memo to get their name in the react dev tools

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://bugzilla.redhat.com/show_bug.cgi?id=2041475

**Analysis / Root cause**: 
The react "compiler" automatically knows component names for functions:

```tsx
ComonentA(props) { return null; }
```

For arrow functions this works only if they are directly assigned to a variable:

```tsx
const CompontentB = (props) => null
```

This doesn't work when the component is wrapped in `React.memo` or when it is wrapped in a `with...` HOC function.

**Solution Description**: 

1. Add `function` keyword to some components which are wrapped with React.memo or `observer` like `PodSet` so that the component name is clear:

```tsx
const CompontentC = React.memo((props) => null) { ... }
```

to

```tsx
const CompontentC = React.memo(function CompontentC(props) { ....
```

For example: `WorkloadPodsNode`, `TopologyListViewComponent`, `TopologyListView`, `AppWithExtensions`, `AppRouter`, `CaptureTelemetry`, `PollConsoleUpdates`, `NamespacesList`, `NamespacesPage`

2. Added an explicit displayName to component wrapper/HOC `withFallback`, `withLastNamespace`, `withserSettings`, etc.

3. Renamed this "hook" to a "utils" because it is called in useEffect (which should not call hooks itself) and in fact it is not a hook.

```diff
-import { useOverviewAlertsUpdater } from '../hooks/useOverviewAlertsUpdater';
-import { useOverviewMetricsUpdater } from '../hooks/useOverviewMetricsUpdater';
+import { subscribeOverviewAlerts } from '../utils/subscribeOverviewAlerts';
+import { subscribeOverviewMetrics } from '../utils/subscribeOverviewMetrics';
```

4. Moved a conditional return before a `useEffect` in `namespace.jsx` so that this conforms to the general hook rules.

**Screenshots of the dev tools**: 

App root components:
<table><tr>
<td>
</td>
<td></td>
</tr></table>

With last namespace:
<table><tr>
<td>
</td>
<td></td>
</tr></table>

WorkloadPodsNode:
<table><tr>
<td>
</td>
<td></td>
</tr></table>

PodSet:
<table><tr>
<td>
</td>
<td></td>
</tr></table>

TopologyListView:
<table><tr>
<td>
</td>
<td></td>
</tr></table>

**Unit test coverage report**: 
Unchanged

**Test setup:**
Check if the components and pages still work.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by nemesis09** [comment]
tested some of the components locally. react dev tools looks much detailed.

**Review by rottencandy — COMMENTED** [review]
Perhaps there is an eslint rule that can help with this?

I found https://eslint.org/docs/rules/func-names
Haven't tested it, but seems to check for named expressions.
Would be really nice if we enabled this, debugging would become more easier.

**Comment by jerolimov** [comment]
> Perhaps there is an eslint rule that can help with this?
> 
> I found https://eslint.org/docs/rules/func-names Haven't tested it, but seems to check for named expressions. Would be really nice if we enabled this, debugging would become more easier.

Yep would be cool. But whats about the current PR? Any issues with adding this code? :smirk:

File: frontend/packages/topology/src/components/list-view/ListElementWrapper.tsx [hunk 1]
```diff
@@ -26,38 +26,44 @@ const ListElementComponent: React.FC<ListElementWrapperProps> = observer(
...
+              onSelect={onSelect}
+              selectedIds={selectedIds}
+            />
+          ))}
+      </>
     );
   },
 );
 
+const ListElementWrapper: React.FC<ListElementWrapperProps> = observer(function({
```
[the remark this answers is not in the document]
Comment (jerolimov) [reply frontend/packages/topology/src/components/list-view/ListElementWrapper.tsx:53]: Nice, added :+1:

**Review by rottencandy — APPROVED** [review]
Thanks! Changes look good.
/lgtm

**Pull request merged**